packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java

```diff
@@ -113,6 +126,37 @@ private static String getDataColumn(
     return null;
   }
 
+  private static String getPathFromRemoteUri(final Context context, final Uri uri) {
+    InputStream inputStream = null;
+    OutputStream outputStream = null;
+    try {
+      inputStream = context.getContentResolver().openInputStream(uri);
+      File file = File.createTempFile("cachedImage", "jpg", context.getCacheDir());
+      outputStream = new FileOutputStream(file);
+      if (inputStream != null) {
+        byte[] buffer = new byte[4 * 1024];
+        int read;
+        while ((read = inputStream.read(buffer)) != -1) {
+          outputStream.write(buffer, 0, read);
+        }
+        outputStream.flush();
+        return file.getPath();
```

Comment:
We are returning a file path here even if there are exceptions thrown while closing the streams. We should only return a file path if all the I/O completes normally.